cfme/web_ui/__init__.py

```diff
@@ -3994,6 +3994,24 @@ def controller_name():
     return sel.execute_script('return ManageIQ.controller;')
 
 
+def match_location(controller=None, title=None, summary=None):
+    """Does exact match of passed data
+
+        Returns:
+        :py:class:`bool`
+    """
+    result = []
+    if controller:
+        result.append(controller_name() == controller)
+    if title:
+        result.append(browser_title() == title)
+    if summary:
+        result.append(summary_title() == summary or
+                      sel.is_displayed('//h3[text() == "{}"]'.format(summary)))
```

Comment:
OK, a few things here,
1) ```'//h3[text() == "{}"]'``` Doesn't yield a legal expression, it actually needs to be ```=```. We have often used ```.``` to match the text and not the ```text()``` function as it has given us problems, ask @mfalesni.
2) At least on the Hosts page, as it was the only one I tried, the summary heading is an ```<h1>``` not an ```<h3>```. We should ask for this to be changed I think, but until then, we should perhaps put in so that it can use either tag.